### PR TITLE
[CI][NFC] Fixed GFX driver update PR

### PR DESCRIPTION
**Pull request by apstasen**
Now generate test matrix correctly use github.event.pull_request.head.sha that is required for pull_request_target.

### CODE REVIEW

File: .github/workflows/sycl_gen_test_matrix.yml [hunk 1]
```diff
@@ -24,6 +24,10 @@ on:
         type: string
         required: false
         default: ${{ github.run_id }}-${{ github.run_attempt }}
+      ref:
+        description: "Commit to use for checkout or github.sha by default"
+        type: string
+        required: false
```
Comment (pvchupin) [inline .github/workflows/sycl_gen_test_matrix.yml:30]: should we have github.sha as default here? and use just inputs.ref below?

Comment (apstasen) [reply .github/workflows/sycl_gen_test_matrix.yml:30]: I'm not sure you can use variable like ${{github.sha}} as default value here. Only constants are possible I think.

Comment (apstasen) [reply .github/workflows/sycl_gen_test_matrix.yml:30]: Looks like you are right and we can use variables in this place after https://github.com/Azure/actions/issues/2 fix

**Review by pvchupin — APPROVED** [review]

File: .github/workflows/sycl_gen_test_matrix.yml (hunk 1 above)
Comment (apstasen) [reply .github/workflows/sycl_gen_test_matrix.yml:30]: P.S. Also noticed I used it before in uniq input param just fine :)

**Pull request merged**